### PR TITLE
Add opt-in to always prompt for repo for issue creation and add comment to issue file specifying the repo

### LINKED ISSUE #2099: Hint which repo an issue will be opened in
Testing https://github.com/microsoft/vscode-pull-request-github/issues/2095

I think it would be helpful if the new issue form editor would tell me which repo the issue is being opened in, just to give me confidence that it got it right.

### PR DESCRIPTION
**Pull request by mikeseese**
This PR adds a `githubIssues.alwaysPromptForNewIssueRepo` config variable that defaults to `false`. When set to `true`, if the user creates a new issue via the `+` button in the side panel, the extensions will always prompt the user for a repo to create the issue for instead of using the current open file. Behavior should not change for creating issues from selection/etc that are supposed to be directly contextual.

This PR also adds a comment in the `NewIssue.md` file telling the user which repo the issue will be created in if it's available:

image

This PR fixes #2099.

### CODE REVIEW

**Comment by mikeseese** [comment]
@microsoft-github-policy-service agree

File: src/issues/issueFeatureRegistrar.ts [hunk 1]
```diff
@@ -964,18 +977,19 @@ export class IssueFeatureRegistrar implements vscode.Disposable {
...
-		const assigneeLine = `${ASSIGNEES} ${assignees && assignees.length > 0 ? assignees.map(value => '@' + value).join(', ') + ' ' : ''
+		const assigneeLine = `${ASSIGNEES} ${options?.assignees && options.assignees.length > 0 ? options.assignees.map(value => '@' + value).join(', ') + ' ' : ''
 			}`;
 		const labelLine = `${LABELS} `;
 		const milestoneLine = `${MILESTONE} `;
 		const projectsLine = `${PROJECTS} `;
 		const cached = this._newIssueCache.get();
-		const text = (cached && cached !== '') ? cached : `${title ?? vscode.l10n.t('Issue Title')}\n
+		const text = (cached && cached !== '') ? cached : `${options?.title ?? vscode.l10n.t('Issue Title')}\n
+${options?.repoUri ? `<!-- ${vscode.l10n.t(`This issue will be created in repo ${options.repoUri.toString()}. Changing this line has no effect.`)} -->\n` : ''}
```
[the remark this answers is not in the document]
Comment (alexr00) [reply src/issues/issueFeatureRegistrar.ts:987]: `options.repoUri.toString()` is going to be a local file location. Instead, we should show the GitHub.com URL.

File: src/issues/issueFile.ts [hunk 2]
```diff
@@ -206,6 +214,24 @@ export async function extractMetadataFromFile(repositoriesManager: RepositoriesM
 	}
 	let assignees: string[] | undefined;
 	text = text.substring(indexOfEmptyLine + 2).trim();
+	if (text.startsWith('<!--')) {
+		text = text.substring(indexOfEmptyLine + 2).trim();
```
Comment (alexr00) [inline src/issues/issueFile.ts:218]: `indexOfEmptyLine` hasn't been updated, this isn't getting a useful substring.

**Review by alexr00 — CHANGES_REQUESTED** [review]
Thanks for the PR! I have some comments.

**Comment by mikeseese** [comment]
@alexr00 It got a little more complicated, but now `origin` will contain information to reproduce the correct folder manager and remote by adding a new `repo` URI scheme.

**Review by LocoabordoR8 — APPROVED** [review]

**Comment by rzhao271** [comment]
Merge conflict
Moving the milestone

**Comment by alexr00** [comment]
Thanks for your patience on this! I'll look at it next week when we start our annual clean up iteration.

**Review by alexr00 — APPROVED** [review]
@mikeseese, I had some significant changes I wanted with regard to the URI related opterations and schemes. Rather than make you go through more back and forth to make changes that are really my own debt, I've pushed the changes to your branch. I plan to merge the PR after we release this week. Thank you for the PR!

**Comment by mikeseese** [comment]
Sounds good, thanks!

**Review by lramos15 — APPROVED** [review]

**Pull request merged**